src/features/features.jl

```diff
@@ -1,29 +1,48 @@
 module FeatureDescriptor
 
-using ..ChemistryFeaturization.AbstractType: AbstractFeatureDescriptor
+using ..ChemistryFeaturization.AbstractType:
+    AbstractAtoms, AbstractFeatureDescriptor, AbstractCodec
 
 import ..ChemistryFeaturization.encodable_elements
 encodable_elements(fd::AbstractFeatureDescriptor) =
     throw(MethodError(encodable_elements, fd))
 export encodable_elements
 
+
+
 import ..ChemistryFeaturization.encode
-encode(fd::AbstractFeatureDescriptor, object_to_be_encoded) = throw(MethodError(encode, fd))
+"""
+    encode(fd::AbstractAtomFeatureDescriptor, atoms::AbstractAtoms)
+Encode `atoms` using the feature descriptor `fd`.
+"""
+encode(fd::AbstractFeatureDescriptor, atoms::AbstractAtoms) = fd(atoms)
 export encode
 
 import ..ChemistryFeaturization.decode
-decode(fd::AbstractFeatureDescriptor, encoded_feature) = throw(MethodError(decode, fd))
+"""
+    decode(fd::AbstractAtomFeatureDescriptor, encoded_feature)
+Decode `encoded_feature` using the feature descriptor `fd`.
+"""
+decode(fd::AbstractFeatureDescriptor, encoded_feature) =
+    fd.encoder_decoder(fd, encoded_feature)
 export decode
 
+(codec::AbstractCodec)(fd::AbstractFeatureDescriptor, a::AbstractAtoms) = error(
+    "Logic specifying how $(typeof(codec))'s encoding mechanism actually encodes $(typeof(fd)) is undefined.",
+)
+(codec::AbstractCodec)(fd::AbstractFeatureDescriptor, encoded_feature) = error(
```

Comment:
Throwing an `error` here, because I'm not sure how to throw a `MethodError` here 😅